pkg/builder/controller.go

```diff
@@ -344,12 +340,15 @@ func (blder *Builder) doController(r reconcile.Reconciler) error {
 	globalOpts := blder.mgr.GetControllerOptions()
 
 	ctrlOptions := blder.ctrlOptions
+	if ctrlOptions.Reconciler != nil && r != nil {
```

Comment:
I think it makes sense to return an error when two reconcilers are set via the buidler and not just silently pick one

(I can move this to a separate PR if preferred)